weaviate/collection/classes.py

```diff
@@ -824,3 +826,50 @@ def to_dict(self) -> Dict[str, Any]:
 
 class Tenant(BaseModel):
     name: str
+
+
+class Filters:
+    pass
+
+
+class FilterAnd(Filters):
+    def __init__(self, *args: Filters):
+        self.filters: List[Filters] = list(args)
+
+    # replace with the following once 3.11 is the minimum version
+    #     Operator: weaviate_pb2.Filters.OperatorType = weaviate_pb2.Filters.OperatorAnd
+    @property
+    def operator(self) -> weaviate_pb2.Filters.OperatorType:
+        return weaviate_pb2.Filters.OperatorAnd
+
+
+class FilterOr(Filters):
+    def __init__(self, *args: Filters):
+        self.filters: List[Filters] = list(args)
+
+    # replace with the following once 3.11 is the minimum version
+    #     Operator: weaviate_pb2.Filters.OperatorType = weaviate_pb2.Filters.OperatorOr
+    @property
+    def operator(self) -> weaviate_pb2.Filters.OperatorType:
+        return weaviate_pb2.Filters.OperatorOr
+
+
+class FilterOperator(Enum):
+    EQUAL = weaviate_pb2.Filters.OperatorEqual
+    NOT_EQUAL = weaviate_pb2.Filters.OperatorNotEqual
+    GREATER_THAN = weaviate_pb2.Filters.OperatorGreaterThan
+    GREATER_THAN_EQUAL = weaviate_pb2.Filters.OperatorGreaterThanEqual
+    LESS_THAN = weaviate_pb2.Filters.OperatorLessThan
+    LESS_THAN_EQUAL = weaviate_pb2.Filters.OperatorLessThanEqual
+    LIKE = weaviate_pb2.Filters.OperatorLike
+    IS_NULL = weaviate_pb2.Filters.OperatorIsNull
+
+
+FilterValues = Union[int, float, str, bool, datetime.date]
+
+
+@dataclass
+class FilterValue(Filters):
+    path: Union[str, List[str]]
```

Comment:
Would naming this property rather than path be more descriptive? Perhaps not since it might confuse the instance where the user is filtering on a reference

Reply:
I had it as property, problem is that property is a reserved name and flake complains

Reply:
😭